resources/files/windows/PuppetBolt/PuppetBoltFunctions.ps1

```diff
@@ -0,0 +1,1639 @@
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $SSHPrivateKey,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $RunAs,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $SudoPassword,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $Concurrency,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $CompileConcurrency,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ModulePath,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $BoltDir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ConfigFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $InventoryFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $SaveRerun,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Transport,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $ConnectTimeout,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $TTY,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Tmpdir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $Color,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltOutputLevel]  $OutputLevel,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $TargetCommand,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $PlanName,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $TaskName,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $BoltCommand,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ScriptLocation,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $PuppetManifest,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Source,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Destination,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Targets,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Query
+    )
+    begin {
+        $_flags = Convert-BoltParams @PSBoundParameters
```

Comment:
nit: Underscore usage is a bit odd.  Underscore names tend to be a ruby thing, not so much PowerShell